runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -109,18 +99,10 @@ protected boolean discoverAddressSpace(final UUID streamId,
 
             // Read the current address
             ILogData d;
-            try {
-                log.trace("followBackpointers: readAddress[{}]", currentAddress);
-                d = read(currentAddress, readStartTime);
-            } catch (TrimmedException e) {
-                if (getReadOptions().isIgnoreTrim()) {
-                    log.warn("followBackpointers: Ignoring trimmed exception for address[{}]," +
-                            " stream[{}]", currentAddress, id);
-                    return !queue.isEmpty();
-                } else {
-                    throw e;
-                }
-            }
+
+            log.trace("followBackpointers: readAddress[{}]", currentAddress);
+            d = read(currentAddress, readStartTime);
```

Comment:
I'm not sure if pure back pointer implementation is compatible with sparse trim, is it? cause a compacted address loses backpointer information, am I right? or it is always kept around?

Reply:
I also think we should eliminate backpointer address discover in the new GC branch.